### ISSUE TITLE
Arrow keys with meta keys don't work as expected

### ISSUE THREAD

**Issue by sarahdayan** [testing-library/user-event#489]
- `@testing-library/user-event` version: 12.2.0
- Testing Framework and version: jest 24.1.0
- DOM Environment: react-dom 16.13.1

**Relevant code or config**

See reproduction below.

**Reproduction repository**

https://codesandbox.io/s/smoosh-violet-el3sh?file=/src/index.test.js

**Problem description**

The library doesn't seem to support the combination of arrow keys and meta keys within an input. For instance, on Chrome, Firefox and Safari, when the cursor is at the end of the text value and you press `Alt + ArrowLeft`, it jumps back one word. When you press `Meta + ArrowLeft`, it jumps back the entire value and sets the cursor at the beginning. Same goes with `ArrowRight`, in the opposite direction.

For example, if I type "hello world" then hit `Alt + ArrowLeft`, my cursor should be right before "there" (index 6). If I hit `Meta + ArrowLeft`, it should be right before "hello" (index 0).

You can try this in the **Browser** tab.

However, the test using `userEvent.type` with `{alt}` or `{meta}` don't pass (see **Tests** tab). It also doesn't work with `fireEvent.keyDown`, but interestingly, the output is different (see failing tests).

I don't know if these combinations are considered standard, but they do work out of the box on the major browsers, so it might be interesting to implement them.

**Comment by kentcdodds** [comment]
The biggest challenge with this kind of behavior is that AFAIK it's not standardized. This means we would have to make a decision of which behavior to simulate. This is not something I think we should do 😬

**Comment by malwilley** [comment]
> I don't know if these combinations are considered standard

As someone who has the (dis)pleasure of switching between macOS and windows/linux daily, I wish this were the case 😄 . `Alt + ArrowLeft` in macOS is equivalent to `Ctrl + ArrowLeft` in windows. And `Meta + ArrowLeft` doesn't really have a parallel (perhaps the `Home` key?).

IMO this probably isn't something we want to get into, but you bring up a good point. There should probably be a discussion on how much behavior this library should be  expected to cover. At the very least, a section in the readme would be helpful for users who might be wondering the same thing.

**Comment by kentcdodds** [comment]
Yeah, I think we take a stand at standardized behavior (as much as possible) and we don't implement non-standard behavior.

**Comment by malwilley** [comment]
Turns out the readme does actually say this already, so ignore my suggestion there:

> We take the same stance as Cypress in that we do not simulate the behavior that happens with modifier key combinations as different operating systems function differently in this regard.

**Comment by nickmccurdy** [comment]
Should this be closed then, or is there a different way we can approach it?

**Comment by sarahdayan** [comment]
Totally understand that this is non-standard behavior, and doesn't belong in the library.

@nickmccurdy In the example I gave, the way to circumvent it is to actually move the cursor manually, either by hitting `ArrowLeft` the right number of times (e.g., `"hello there{arrowleft}{arrowleft}{arrowleft}{arrowleft}{arrowleft}{backspace}"`) or by setting the cursor with `HTMLInputElement.setSelectionRange` and triggering a click event.

Closing it, thanks for the discussion!
